File: src/03.FrontEnd/04.FrontEnd.Logics/Modules/Administration/Configurations/GetConfigurations/GetConfigurationsQuery.cs
using SolutionTemplate2.Shared.Dto.Modules.Administration.Configurations.GetConfigurations;

namespace SolutionTemplate2.FrontEnd.Logics.Modules.Administration.Configurations.GetConfigurations;

public sealed record GetConfigurationsQuery : IRequest<GetConfigurationsResponse>
{
}

public sealed class GetConfigurationsQueryHandler(IBackEndApiService backEndApiService)
    : IRequestHandler<GetConfigurationsQuery, GetConfigurationsResponse>
{
    public async Task<GetConfigurationsResponse> Handle(GetConfigurationsQuery request, CancellationToken cancellationToken)
    {
        var restRequest = new RestRequest(GetConfigurationsRoute.Pattern, Method.Get);

        return await backEndApiService.SendRequestAsync<GetConfigurationsResponse>(restRequest, cancellationToken);
    }
}

File: src/04.FrontEndReact/04.FrontEndReact.Logics/Modules/Administration/Configurations/GetConfigurations/GetConfigurationsQuery.ts
import type { GetConfigurationsResponse } from "@services/Dto/Modules/Administration/Configurations/GetConfigurations/GetConfigurationsResponse";
import { GetConfigurationsRoute } from "@services/Dto/Modules/Administration/Configurations/GetConfigurations/GetConfigurationsRoute";
import { getBackEndApiService } from "@infrastructure/BackEndApi/ConfigureBackEndApi";

export async function sendGetConfigurationsQuery(): Promise<GetConfigurationsResponse> {
  return getBackEndApiService().sendRequestAsync<GetConfigurationsResponse>({
    resourceUri: GetConfigurationsRoute.ResourceUri,
    method: "GET",
  });
}
